assets/js/components/game.js

```diff
@@ -0,0 +1,137 @@
+      for (let m = 0; m < winCombination.length; m += 1) {
+        console.log(isEqual(players[k].playersMove, winCombination[m]));
+        if (isEqual(players[k].playersMove, winCombination[m])) {
+          console.log('I am from winner');
+          return players[k];
+        }
+      }
+    }
+    return false;
+  };
+  const declareResult = () => {
+    document.querySelector('.status').style.display = 'block';
+    if (winner()) {
+      document.querySelector('.result').innerText = `Winner is: ${winner().name}`;
+    } else if (board.boardIsFull()) {
+      document.querySelector('.result').innerText = 'It is a Draw';
+    }
+  };
+  const checkResult = () => {
+    if (winner() || board.boardIsFull()) {
+      result = true;
+      declareResult();
+    }
+  };
+  const fillCell = (element) => {
+    if (!result) {
+      if (turn === 0) {
+        if (board.cellIsFilled(element)) {
+          document.getElementById(element).innerText = 'X';
+          document.querySelector('.player-0-panel').classList.remove('border-danger');
+          document.querySelector('.player-1-panel').classList.add('border-danger');
+          board.setCell(element, 'X');
+          players[0].playersMove.push(element);
+          turn = 1;
+        }
+      } else if (turn === 1) {
+        if (board.cellIsFilled(element)) {
+          document.getElementById(element).innerText = 'O';
+          document.querySelector('.player-1-panel').classList.remove('border-danger');
+          document.querySelector('.player-0-panel').classList.add('border-danger');
+          board.setCell(element, 'O');
+          players[1].playersMove.push(element);
+          turn = 0;
+        }
+      }
+    }
+    checkResult();
+  };
+  const loadGame = () => {
+    players = [Player(playerXName, 'X'), Player(playerOName, 'O')];
+  };
+  const showGrid = () => {
+    initialTarget.innerHTML = display.createTable()
+                              + display.createUserInfo(playerXName, playerOName);
+    // eslint-disable-next-line no-use-before-define
+    document.querySelector('.reset').addEventListener('click', reset);
+    // eslint-disable-next-line no-use-before-define
+    document.querySelector('.new-game').addEventListener('click', newGame);
+    const cells = document.querySelectorAll('.cell');
+    for (let i = 0; i < cells.length; i += 1) {
+      cells[i].addEventListener('click', () => {
+        fillCell(i);
+      });
+    }
+  };
+  const showGameBoard = () => {
+    const initialView = document.querySelector('.initial-view');
+    initialTarget.removeChild(initialView);
+    showGrid();
+  };
+  const setGameView = () => {
+    showGameBoard();
+    loadGame();
+  };
+  const addPlayEvent = (firstLook) => {
+    initialTarget.innerHTML = firstLook;
+    const playButton = document.querySelector('#play');
+    playButton.addEventListener('click', () => {
+      playerXName = document.querySelector('#exampleInputName1').value;
```

Comment:
All `dom` related task should be done separately in another file say as I've specified earlier.